File: frontend/src/types/Polymarket.ts
export interface PolymarketOdds {
    clobTokenIds: string;
    id: string
    outcomes: string
    outcomePrices: string
    volumeNum: number
    enableOrderBook: boolean
    gameStartTime: string
    question: string
    slug: string
    bestAsk: number;
    orderMinSize: number;
    volume: string
    conditionId: string
}

export interface CleanedPolymarketOdds extends Omit<PolymarketOdds, 'outcomes' | "clobTokenIds">{
    outcomes: string[]
    clobTokenIds: string[]
}

File: backend/models/polymarket.py
from typing import List, Optional

from pydantic import BaseModel, RootModel


# PolymarketOdds Pydantic model
class PolymarketOdds(BaseModel):
    clobTokenIds: str
    id: str
    outcomes: str
    outcomePrices: str
    volumeNum: Optional[float] = None
    enableOrderBook: bool
    gameStartTime: Optional[str] = None
    question: str
    slug: str
    bestAsk: float
    orderMinSize: float
    volume: Optional[str] = None
    conditionId: str

    def __hash__(self):
        return hash(self.id)


class PolymarketResponse(RootModel):
    root: List[PolymarketOdds]


class CleanedPolymarketOdds(BaseModel):
    id: str
    outcomes: List[str]
    outcomePrices: List[str]
    volumeNum: float
    enableOrderBook: bool
    gameStartTime: str | None
    question: str
    slug: str
    bestAsk: float
    orderMinSize: float
    volume: str
    conditionId: str
    clobTokenIds: List[str]
